src/url.js

```diff
@@ -123,6 +123,15 @@ export function parseUrl(url) {
 }
 
 /**
+ * Duplicate of parseUrl, but return a cloned version of the returned obj.
+ * @param {string} url
+ * @return {!Location}
+ */
+export function parseUrlAndClone(url) {
+  return Object.assign({}, parseUrl(url));
```

Comment:
Can we actually use `Object.assign`?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

We do in a few places, but it might be problematic with no Safari 8 support.